src/ray/raylet/pubsub/pubsub_client.cc

```diff
@@ -0,0 +1,161 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/raylet/pubsub/pubsub_client.h"
+
+namespace ray {
+
+void PubsubClient::SubcribeObject(
+    const rpc::Address &owner_address, const ObjectID &object_id,
+    SubscriptionCallback subscription_callback,
+    SubscriptionFailureCallback subscription_failure_callback) {
+  rpc::Address subscriber_address;
+  subscriber_address.set_raylet_id(self_node_id_.Binary());
+  subscriber_address.set_ip_address(self_node_address_);
+  subscriber_address.set_port(self_node_port_);
+  // Make a long polling connection if we never made the one with this owner for pubsub
+  // operations.
+  const auto owner_worker_id = WorkerID::FromBinary(owner_address.worker_id());
+  if (subscription_map_.count(owner_worker_id) == 0) {
+    subscription_map_.emplace(owner_worker_id, SubscriptionInfo(owner_address));
+    MakeLongPollingPubsubConnection(owner_address, subscriber_address);
+  }
+  auto subscription_it = subscription_map_.find(owner_worker_id);
+  RAY_CHECK(subscription_it != subscription_map_.end());
```

Comment:
Same suggestion about using `find()` from the start:
```suggestion
  auto subscription_it = subscription_map_.find(owner_worker_id);
  if (subscription_it == subscription_map_.end()) {
    subscription_it = subscription_map_.emplace(owner_worker_id, SubscriptionInfo(owner_address)).first;
    MakeLongPollingPubsubConnection(owner_address, subscriber_address);
  }
```